src/lib/tls/tls_channel.h

```diff
@@ -32,22 +32,58 @@ class Handshake_Message;
 class BOTAN_DLL Channel
    {
    public:
-      typedef std::function<void (const byte[], size_t)> output_fn;
-      typedef std::function<void (const byte[], size_t)> data_cb;
-      typedef std::function<void (Alert, const byte[], size_t)> alert_cb;
-      typedef std::function<bool (const Session&)> handshake_cb;
-      typedef std::function<void (const Handshake_Message&)> handshake_msg_cb;
-
-      Channel(output_fn out,
-              data_cb app_data_cb,
-              alert_cb alert_cb,
-              handshake_cb hs_cb,
-              handshake_msg_cb hs_msg_cb,
+      static size_t IO_BUF_DEFAULT_SIZE;
+
+      class Callbacks
+         {
+            public:
+               typedef std::function<void (const byte[], size_t)> output_fn;
+               typedef std::function<void (const byte[], size_t)> data_cb;
+               typedef std::function<void (Alert, const byte[], size_t)> alert_cb;
```

Comment:
There is no reason for alert to have the byte[]/size_t arguments here. They were previously used for heartbeat data, and the heartbeat function has been removed. The only reason they are still used in the alert callback signature is for API compatibility, but that concern is moot given the other changes.